### PR TITLE
Add admin, presenter, publisher request and publisher from listener lists into broadcast object

### PR DESCRIPTION
**Pull request by mustafaboleken**
https://github.com/ant-media/Ant-Media-Server/issues/6147

### CODE REVIEW

File: src/main/java/io/antmedia/datastore/db/types/Broadcast.java [hunk 1]
```diff
@@ -207,6 +207,18 @@ public class Broadcast {
 	@ApiModelProperty(value = "Initial time to start playing. It can be used in VoD file or stream sources that has seek support")
 	private long seekTimeInMs = 0;
 
+	@ApiModelProperty(value = "The list of the play only users who requested to publish the stream. It is used in conference scenario.")
+	List<String> publisherRequestList = new ArrayList<String>();
```
Comment (mekya) [inline src/main/java/io/antmedia/datastore/db/types/Broadcast.java:211]: These are the subsriberId of the user or they are stream id?

File: src/main/java/io/antmedia/datastore/db/types/Broadcast.java [hunk 2]
```diff
@@ -207,6 +207,18 @@ public class Broadcast {
...
+
+	@ApiModelProperty(value = "The list of the users who are approved to publish the stream. It is used in conference scenario.")
+	List<String> publisherFromListenerList = new ArrayList<String>();
```
Comment (mekya) [inline src/main/java/io/antmedia/datastore/db/types/Broadcast.java:214]: Do we need to have different fields for user scenarios? I mean publisherFromListenerList seems like scenario specific to me.

File: src/main/java/io/antmedia/datastore/db/types/Broadcast.java [hunk 3]
```diff
@@ -207,6 +207,18 @@ public class Broadcast {
...
+
+	@ApiModelProperty(value = "The list of the presenter users. It is used in conference scenario.")
+	List<String> presenterList = new ArrayList<String>();
+
+	@ApiModelProperty(value = "The list of the admin users. It is used in conference scenario.")
+	List<String> adminList = new ArrayList<String>();
```
Comment (mekya) [inline src/main/java/io/antmedia/datastore/db/types/Broadcast.java:220]: These are the subscriberIds?

File: src/main/java/io/antmedia/datastore/db/types/Broadcast.java [hunk 4]
```diff
@@ -207,6 +207,18 @@ public class Broadcast {
...
 	private long seekTimeInMs = 0;
 
+	@ApiModelProperty(value = "The list of the play only users who requested to publish the stream. It is used in conference scenario.")
+	List<String> publisherRequestList = new ArrayList<String>();
+
+	@ApiModelProperty(value = "The list of the users who are approved to publish the stream. It is used in conference scenario.")
+	List<String> publisherFromListenerList = new ArrayList<String>();
+
+	@ApiModelProperty(value = "The list of the presenter users. It is used in conference scenario.")
+	List<String> presenterList = new ArrayList<String>();
```
Comment (mekya) [inline src/main/java/io/antmedia/datastore/db/types/Broadcast.java:217]: presenterList is also the subtracks of this stream. Do we need to have additional field?

File: src/main/java/io/antmedia/datastore/db/DataStore.java [hunk 5]
```diff
@@ -1238,29 +1256,93 @@ else if (sortBy.equals("endDate")) {
...
 	 */
 	public abstract boolean deleteP2PConnection(String streamId);
 
+	/**
+	 * Add a stream into publisher request list
+	 * @param mainTrackId - main track id
+	 * @param streamId - stream id
+	 * @return boolean - success
+	 */
+	public abstract boolean addIntoPublisherRequestList(String mainTrackId, String streamId);
```
Comment (mekya) [inline src/main/java/io/antmedia/datastore/db/DataStore.java:1283]: I think we don't need to additional methods for the new fields, we can just use the updateBroadcast method

**Review by mekya — CHANGES_REQUESTED** [review]
Thank you @mustafaboleken for the implementation. 

I see some opportunities to make it more simple. Here are my ideas, we can discuss it later

- No need to have new abstract methods in DataStore. updateBroadcast method is enough
-  No need to have `presenterList` field in broadcast object. They are subtracks of the main stream and there is a field for subtracks. 
- No need to have publisherFromListener because they are again subtracks.
- Rename publisherRequestList to `subtrackRequestList` or something similar

**Comment by mustafaboleken** [comment]
@mekya thank you for your review. I understand your points but for some scenarios, it's hard to keep track of subtracts. I will  ask them in a meeting.

**Comment by mekya** [comment]
WILL NOT MERGED and we provide another resolution: Role based streaming

**Pull request closed without merging**